test/cpu/test-i386.c

```diff
@@ -880,7 +880,7 @@ void test_fcvt(double a)
 
     fa = a;
     la = a;
-    printf("(float)%f = %f\n", a, fa);
+    printf("(float)%.15e = %.7e\n", a, fa);
```

Comment:
Was it necessary to change to e?

Reply:
I've never really done much formatting of floats, other than the usual 3 decimal places (%0.3f) etc. I wanted to display all values to the limit of their type's precision.
For example imagine a problem float value 'fa = -100000001999999991808.0'. How should I format that to cut off after 7 significant figures, and cope with a very small value say '0.33333333333'? With '%e' it's as easy as '%.7e', => '-1.0000000e+20', and '3.3333333e-01'. But like I say, I know no better.